src/components/input/ButtonBase.js

```diff
@@ -56,8 +58,19 @@ const ButtonBaseNext = function ButtonBase({
     <button
       role={role ?? 'button'}
       {...ariaProps}
+      data-base-component="ButtonBase"
+      /* data-component will be overwritten unless this component is used directly */
+      data-component="ButtonBase"
```

Comment:
Does this make sense?

`IconButton` renders a `ButtonBase` but provides its own `data-component` value, e.g.:

`<button data-component="IconButton" data-base-component="ButtonBase" ... />`

But if you use `ButtonBase` directly (i.e. for a custom button), you'd get:

`<button data-component="ButtonBase" data-base-component="ButtonBase" />`

Reply:
I think `data-component` alone is fine here, since we're dealing with just one element. The main value in these attributes is being able to see which elements come from which components when looking at the DOM tree.

Reply:
Right, but because we're introducing this new/first abstraction in which some of the applied CSS styles for `Button`, `IconButton`, e.g. come from `ButtonBase`, I think there's some value in having a `data-base-component` so that the developer locks on to the idea that some of the classes they're seeing there in the DOM inspector are coming from `ButtonBase`. I feel like it's one little extra helper to give context.